File: physicMentor/dotnetapp/Models/PhysicalTraining.cs
using System.ComponentModel.DataAnnotations;

namespace dotnetapp.Models
{
    public class PhysicalTraining
    {
        public int PhysicalTrainingId { get; set; }

        [Required(ErrorMessage = "Training name is required")]
        public string TrainingName { get; set; } // E.g., Strength Training, Yoga, Pilates

        [Required(ErrorMessage = "Description is required")]
        public string Description { get; set; } // Description of the training

        [Required(ErrorMessage = "Trainer name is required")]
        public string TrainerName { get; set; } // Name of the trainer leading the session

        [Required(ErrorMessage = "Location is required")]
        public string Location { get; set; } // Physical location, e.g., Gym, Park, Studio

        [Required(ErrorMessage = "Indoor/Outdoor information is required")]
        public bool IsIndoor { get; set; } // True for indoor training, false for outdoor

        [Required(ErrorMessage = "Fee is required")]
        [Range(0, double.MaxValue, ErrorMessage = "Fee must be a non-negative value")]
        public decimal Fee { get; set; } // Cost for attending the training

        [Required(ErrorMessage = "Focus area is required")]
        public string FocusArea { get; set; } // Specific physical focus (e.g., cardio, strength, flexibility)

        [Required(ErrorMessage = "Physical requirements are required")]
        public string PhysicalRequirements { get; set; } // Any physical requirements or prerequisites for participation

    }
}

File: physicMentor/springapp/src/main/java/com/examly/springapp/model/PhysicalTraining.java
package com.examly.springapp.model;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Column;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@Entity
public class PhysicalTraining {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long physicalTrainingId;

    @NotBlank(message = "Training name is required")
    private String trainingName;

    @NotBlank(message = "Description is required")
    private String description;

    @NotBlank(message = "Trainer name is required")
    private String trainerName;

    @NotBlank(message = "Location is required")
    private String location;

    @NotNull(message = "Indoor/Outdoor information is required")
    private Boolean isIndoor;

    @NotNull(message = "Fee is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Fee must be a non-negative value")
    private Double fee;

    @NotBlank(message = "Focus area is required")
    private String focusArea;

    @NotBlank(message = "Physical requirements are required")
    private String physicalRequirements;

    public PhysicalTraining() {
    }

    // Getters and Setters

    public Long getPhysicalTrainingId() {
        return physicalTrainingId;
    }

    public void setPhysicalTrainingId(Long physicalTrainingId) {
        this.physicalTrainingId = physicalTrainingId;
    }

    public String getTrainingName() {
        return trainingName;
    }

    public void setTrainingName(String trainingName) {
        this.trainingName = trainingName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getTrainerName() {
        return trainerName;
    }

    public void setTrainerName(String trainerName) {
        this.trainerName = trainerName;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public Boolean getIsIndoor() {
        return isIndoor;
    }

    public void setIsIndoor(Boolean isIndoor) {
        this.isIndoor = isIndoor;
    }

    public Double getFee() {
        return fee;
    }

    public void setFee(Double fee) {
        this.fee = fee;
    }

    public String getFocusArea() {
        return focusArea;
    }

    public void setFocusArea(String focusArea) {
        this.focusArea = focusArea;
    }

    public String getPhysicalRequirements() {
        return physicalRequirements;
    }

    public void setPhysicalRequirements(String physicalRequirements) {
        this.physicalRequirements = physicalRequirements;
    }
}